ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -1624,7 +1626,7 @@ private InstallSnapshotReplyProto notifyStateMachineToInstallSnapshot(
       if (isSnapshotNull.compareAndSet(true, false)) {
         LOG.info("{}: InstallSnapshot notification result: {}", getMemberId(),
             InstallSnapshotResult.SNAPSHOT_UNAVAILABLE);
-        inProgressInstallSnapshotRequest.compareAndSet(firstAvailableLogTermIndex, null);
+        inProgressInstallSnapshotRequest.compareAndSet(firstAvailableLogTermIndex.hashCode(), 0);
```

Comment:
The hasCode() would still be different between to iterations of this call as we create a firstAvailableLogTermIndex object every time in notifyStateMachineToInstallSnapshot(). 
Instead, we could set inProgressInstallSnapshotRequest to firstAvailableLogIndex. This will be a long value which will not change for different iterations of the same request.

Reply:
looks good , i will do this in a new commit